TypeScript error: mcp.ts passes zod v3 compat ZodObject to z4mini.toJSONSchema()

## Bug

`initializers/mcp.ts` fails type-checking (`tsc`) when consumed by a project that depends on keryx with zod v3 compat.

### Error

```
keryx/initializers/mcp.ts(465,27): error TS2769: No overload matches this call.
  Overload 1 of 2, '(schema: $ZodType<...>, _params?: ToJSONSchemaParams | undefined): JSONSchema':
    Argument of type 'ZodObject<{ [x: string]: any; }, "strip", ZodTypeAny, ...>' is not assignable
    to parameter of type '$ZodType<unknown, unknown, $ZodTypeInternals<unknown, unknown>>'.
      Property '_zod' is missing in type 'ZodObject<...>' but required in type '$ZodType<...>'.
  Overload 2 of 2, '(registry: $ZodRegistry<...>, _params?: RegistryToJSONSchemaParams): { ... }':
    ...missing properties: _meta, _schema, _map, _idmap, and 5 more.
```

### Root cause

In `mcp.ts`, keryx constructs a zod object using `z.object(...)` (imported from bare `"zod"`) and passes it to `z4mini.toJSONSchema()` (imported from `"zod/v4-mini"`):

```typescript
// mcp.ts lines 5-6
import { z } from "zod";
import * as z4mini from "zod/v4-mini";

// mcp.ts line 465
z4mini.toJSONSchema(z.object({ [key]: fieldSchema }), {
  target: "draft-7",
  io: "input",
});
```

When a consumer depends on `"zod": "^3.24.0"`:
1. The bare `"zod"` import resolves to `zod/v3/external` (the v3 compat layer)
2. `z.object(...)` returns a v3-compat `ZodObject` type
3. `z4mini.toJSONSchema()` expects a v4-native `$ZodType` with the `_zod` property
4. These types are structurally incompatible — v3 compat `ZodObject` doesn't have `_zod`

At **runtime** this works fine because zod v3 compat objects are actually v4 objects under the hood. But the **TypeScript types** are incompatible, so `tsc` fails.

### Reproduction

1. Create a project with `"keryx": "^0.3.5"` and `"zod": "^3.24.0"`
2. Run `tsc` with `skipLibCheck: true`
3. Observe the error above

### Suggested fix

Either:
- **Option A**: Use `z4mini.object(...)` (or import `z` from `"zod/v4"`) instead of bare `"zod"` for the schema construction on line 465, so both the schema and `toJSONSchema` use v4-native types
- **Option B**: Cast the schema: `z4mini.toJSONSchema(z.object({...}) as any, ...)`
- **Option C**: Require consumers to use zod v4 via `peerDependencies`
- **Option D**: Ship compiled `.d.ts` so `skipLibCheck` covers keryx internals

Option A is cleanest — the function `sanitizeZodSchemaForMCP` already uses `z4mini.toJSONSchema`, so constructing the test object with v4-mini types too would be consistent.

### Environment

- keryx: 0.3.5
- zod (consumer): ^3.24.0 (resolves to 3.25.76)
- zod (keryx): ^4.3.6
- bun: 1.3.9
- typescript: 5.7.x

--- a/packages/keryx/initializers/mcp.ts
+++ b/packages/keryx/initializers/mcp.ts
@@ -2,7 +2,6 @@ import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
 import { WebStandardStreamableHTTPServerTransport } from "@modelcontextprotocol/sdk/server/webStandardStreamableHttp.js";
 import colors from "colors";
 import { randomUUID } from "crypto";
-import { z } from "zod";
 import * as z4mini from "zod/v4-mini";
 import { api, logger } from "../api";
 import { Connection } from "../classes/Connection";
@@ -462,16 +461,16 @@ function sanitizeSchemaForMcp(schema: any): any {
     schema.shape as Record<string, any>,
   )) {
     try {
-      z4mini.toJSONSchema(z.object({ [key]: fieldSchema }), {
+      z4mini.toJSONSchema(z4mini.object({ [key]: fieldSchema }), {
         target: "draft-7",
         io: "input",
       });
       newShape[key] = fieldSchema;
     } catch {
       needsSanitization = true;
-      newShape[key] = z.string().describe(`${key}`);
+      newShape[key] = z4mini.string();
     }
   }
 
-  return needsSanitization ? z.object(newShape) : schema;
+  return needsSanitization ? z4mini.object(newShape) : schema;
 }

